### PR TITLE
Feature/116023 connect recurrence data

### PR DESCRIPTION
**Pull request by paulmskim**
Connect recurrence data and change handlers to UI.

### CODE REVIEW

File: plugins/events-pro/src/modules/data/blocks/recurring/constants.js [hunk 1]
```diff
@@ -91,6 +31,21 @@ export const ON_LABEL = __( 'On', 'events-gutenberg' );
...
+export const DATE = 'date';
+export const COUNT = 'count';
+
+export const LIMIT_TYPE_MAPPING_TO_STATE = {
+	[ ON ]: DATE,
+	[ AFTER ]: COUNT,
+	[ NEVER ]: null,
+};
+
+export const LIMIT_TYPE_MAPPING_FROM_STATE = {
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/data/blocks/recurring/constants.js]: How about lodash `invert` here?

File: plugins/events-pro/src/modules/elements/recurrence-type-picker/template.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/**
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/elements/recurrence-type-picker/template.js]: Paul can you pull latest `task/recurrence` in if it's not already? I renamed this over to `element/type-picker`

File: plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js [hunk 3]
```diff
@@ -1,62 +1,61 @@
...
-	onRecurringMultiDayChange: PropTypes.func,
-	recurringMultiDay: PropTypes.oneOf( options.RECURRING_MULTI_DAY_OPTIONS ),
+const onEndTimeClick = ( dispatchProps, ownProps ) => ( value, onClose ) => {
+	const isAllDay = value === 'all-day';
+	const payload = {};
+
+	if ( isAllDay ) {
+		payload.all_day = true;
+		payload.start_time = '00:00:00';
+		payload.end_time = '23:59:00';
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: should this be `23:59:59`?

Comment (paulmskim) [reply plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: not sure how recurrence currently does this, we've been using minutes as the most granular unit for the rest of the plugin..

Comment (nealfennimore) [reply plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: @bordoni any thoughts on this?

File: plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js [hunk 4]
```diff
@@ -1,62 +1,70 @@
...
+	actions,
+	options,
+	selectors
+} from '@moderntribe/events-pro/data/blocks/recurring';
+import { withStore } from '@moderntribe/common/hoc';
+
+const { KEY_END_TIME } = constants;
+
+const getRecurringMultiDay = ( state, ownProps ) => {
+	// TODO: remove this once we have recurring multi day in state
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: Do we still need this?

File: plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js [hunk 5]
```diff
@@ -1,62 +1,70 @@
...
+	dispatch( actions.editRule(
+		ownProps.index,
+		{ [ KEY_END_TIME ]: value },
+	) );
+	onClose();
 };
 
-export default RecurringToDateTimePicker;
+const onRecurringMultiDayChange = ( dispatch, ownProps ) => ( selectedOption ) => (
+	// TODO: fix this once we have recurring multi day in state
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: Same :question: here

File: plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js (hunk 4 above)
Comment (paulmskim) [reply plugins/events-pro/src/modules/elements/recurring-to-date-time-picker/element.js]: yes, this is a different thing from `multi_day`. I've noted it here: https://gist.github.com/nealfennimore/4e5ab05f3e297eb1017701d3eff9e150#gistcomment-2729938

**Pull request merged**